lib/run_loop/simctl.rb

```diff
@@ -207,6 +207,49 @@ def wait_for_shutdown(device, timeout, delay)
       in_state
     end
 
+    def boot(device)
+      if simulator_state_as_int(device) == SIM_STATES["Booted"]
+        RunLoop.log_debug("Simulator is already shutdown")
+        true
+      else
+        cmd = ["simctl", "boot", device.udid]
+        hash = shell_out_with_xcrun(cmd, DEFAULTS)
+
+        exit_status = hash[:exit_status]
+        if exit_status != 0
+          if simulator_state_as_int(device) == SIM_STATES["Booted"]
+            RunLoop.log_debug("simctl boot called when state is 'Booted'; ignoring error")
```

Comment:
No. The state was not booted when this method was called, you just checked that :-)

Maybe "Simulator online; ignoring problems from 'simctl boot':\n#{hash[:out]}"

Reply:
Actually, this happens occasionally. 

Something external puts the device in the state before our `boot` or `shutdown` can move the device to that state.  We only care that by the end of this method, the device is in the desired state.

Reply:
Done.